federated_content_connector/filters/pipeline.py

```diff
@@ -38,3 +40,36 @@ def run_filter(self, course_key, course_home_url):  # pylint: disable=arguments-
                 course_home_url = getattr(settings, 'EXEC_ED_LANDING_PAGE', EXEC_ED_LANDING_PAGE)
 
         return {'course_key': course_key, 'course_home_url': course_home_url}
+
+
+class CreateIsStartedForCourseStep(PipelineStep):
+    """
+    Step that modifies the isStarted field for the course.
+
+    Example usage:
+
+    Add the following configurations to your configuration file:
+
+        "OPEN_EDX_FILTERS_CONFIG": {
+            "org.openedx.learning.course.is.started.creation.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "federated_content_connector.filters.pipeline.CreateIsStartedForCourseStep"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, course_key, is_started):  # pylint: disable=arguments-differ
+        """
+        Pipeline step that modifies the course home url for externally hosted courses
+        """
+        try:
+            course_details = CourseDetails.objects.get(id=course_key)
+            start_date = course_details.start_date
+            if start_date and start_date <= timezone.now():
+                is_started = True
+        except CourseDetails.DoesNotExist:
+            pass
+
+        return {'course_key': course_key, 'is_started': is_started}
```

Comment:
in case of `DoesNotExist` what will be the value of `is_started`? I think in that case we will get `referenced before assignment`?